### ISSUE TITLE
bug: tfsec doesn't handle symlinks where target file name is different to source

### ISSUE THREAD

**Issue by nirojan** [aquasecurity/tfsec#1595]
**Describe the bug**

**To Reproduce**
1. Create a file called `variables.tf` with following contents.
```hcl
variable "test" {
  default = "test"
}
```
2. Create a folder called `layer_1`
3. Create a symlink to `variables.tf` file outside the `layer_1` folder with different target name by running the following command 
```sh
ln -s ../variables.tf vars.tf
```
4. Ensure the folder structure looks something like below
```sh
❯ tree
.
├── layer_1
│   └── vars.tf -> ../variables.tf
└── variables.tf
```
5. Run tfsec inside `layer_1` folder

**Expected behavior**

tfsec runs its checks and generate a report with findings.

**Screenshots/Output**

```
Scan error: open /tmp/repo/layer_1/variables.tf: no such file or directory
```

**System Info**

 - tfsec version: `v1.8.0`
 - terraform version: `0.15.3`
 - OS: `macOS 12.3 (Intel)`

**Additional context**

**Comment by liamg** [comment]
Looking at this now.

**Comment by liamg** [comment]
Thanks for the report, this should be resolved in `v1.9.1` (building now).

**Comment by nirojan** [comment]
@liamg I can confirm that it is fixed in `v1.9.1`. Thank you for the quick response and fix

**Comment by ArnyDnD** [comment]
@liamg I believe this issue has regressed, at least in `v1.19.1` which is the one I am using.

This is my current setup:
```
$ tree terraform/
terraform/
└── environments
    ├── shared
    │   ├── main.tf
    │   ├── outputs.tf
    │   └── variables.tf
    └── testing
        ├── example.tf
        ├── main.tf -> ../shared/main.tf
        ├── outputs.tf -> ../shared/outputs.tf
        ├── testing.auto.tfvars
        └── variables.tf -> ../shared/variables.tf
```

Executing tfsec on the `terraform/environments/testing` directory with the `--debug` flag I find that tfsec does not catch the symlinks:
```
Paths: args=[terraform/environments/testing/] dir=/code/terraform/environments/testing root=/ rel=code/terraform/environments/testing
[scan:terraform:May  6 09:44:43.955] scanning [&{/ /}] at code/terraform/environments/testing

[rego:May  6 09:44:43.957] Loaded 4 embedded libraries (without embedded policies).

[rego:May  6 09:44:43.960] Loaded 4 embedded libraries.

[rego:May  6 09:44:43.985] Loaded 62 embedded policies.

[scan:terraform:May  6 09:44:44.072] Scanning root module 'code/terraform/environments/testing'...

[debug:parse][root] Parsing 'code/terraform/environments/testing/example.tf'...    <---- only example.tf is captured
[debug:parse][root] Added file code/terraform/environments/testing/example.tf.
[debug:parse][root] Added directory code/terraform/environments/testing.
[debug:parse][root] Read 1 block(s) and 0 ignore(s) for module 'root' (2 file[s])...
[debug:parse][root] Added 5 variables from tfvars.
[debug:parse][root] Error loading module metadata: open //code/terraform/environments/testing/.terraform/modules/modules.json: no such file or directory.
[debug:eval][root] Finished processing 0 submodule(s).
[debug:parse][root] Finished parsing module 'root'.
[rego:May  6 09:44:44.074] Scanning 1 inputs...
```

Please let me know if you need any further information or if I shall create a new issue instead :)

Kindest regards
